syntax = "proto3";
package com.yimuzn.featuresdetection.device.grpc;


service DetectService {
    // 软件给算法发送图片，返回检测到的特征数量
    rpc DetectFeaturesYolo (DetectRequest) returns (DetectResponse);
}


message DetectRequest {
  repeated string imagePath = 1;
}

message FeatureCount {
  // 特征名称
  int32 feature_name = 1;
  // 特征数量
  int32 count = 2;
}


message DetectResponse {
     map<int32, int32> feature_counts = 1;

}



